services/director-v2/src/simcore_service_director_v2/utils/computations.py

```diff
@@ -107,17 +107,23 @@ async def find_deprecated_tasks(
     catalog_client: CatalogClient,
 ) -> list[ServiceKeyVersion]:
 
-    task_services = await logged_gather(
+    services_details = await logged_gather(
         *(
             catalog_client.get_service(
                 user_id=user_id,
                 service_key=key_version.key,
                 service_version=key_version.version,
                 product_name=product_name,
             )
-            for key_version in task_key_versions
+            for key_version in set(task_key_versions)
```

Comment:
Q: Out of curiosity: Did you noticed that we were calling repeated versions or this is a precaution to avoid doing unnecessary calls?

Reply:
well I notice that starting a 60 sleepers:2.0.2 study, it took several seconds to check, where now it takes <1 second